packages/smart-contracts/src/contracts/ERC20TransferrableReceivable.sol

```diff
@@ -0,0 +1,173 @@
+// Contract based on [https://docs.openzeppelin.com/contracts/3.x/erc721](https://docs.openzeppelin.com/contracts/3.x/erc721)
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import '@openzeppelin/contracts/utils/Counters.sol';
+import '@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol';
+
+contract ERC20TransferrableReceivable is ERC721URIStorage {
```

Comment:
(if the `minter = issuer` solution does not fit)
Can we map the `receivableId` to the `paymentReference` instead of the `requestId`?

(if we go with the `minter = issuer` solution)
Can we map the `receivableId` to a hash of (`paymentReference` + `originalPaymentAddress`)  instead?

In both cases, we should completely skip the use of `requestId` at any stage (minting, payment).

Reply:
As discussed, we will only be storing requestId in the metadata of each token. I did want to clarify that in this contract, none of the `requestTokenId` variables are actually the ones used by Request Network. They're purely for managing the ERC721 tokens.